Makefile

```diff
@@ -32,13 +33,8 @@ GITHUB_ACCESS_TOKEN := $(shell cat /builder/home/git-token 2> /dev/null)
 PEGOMOCK_PACKAGE := github.com/petergtz/pegomock/
 CGO_ENABLED = 0
 
-all: build
-full: check
-check: build test
-
-version:
 ifeq (,$(wildcard pkg/version/VERSION))
-TAG := $(shell git fetch --all -q 2>/dev/null && git describe --abbrev=0 --tags 2>/dev/null)
+TAG := $(shell git describe --abbrev=0 --tags 2>/dev/null)
```

Comment:
FYI, I am removing the `git fetch` here. It takes a lot of time and I don't think is needed. For sure not during development. 

If it is an issue on CI or during release I prefer to address this somewhere else.

Reply:
If we want to do the setting of `VERSION` conditionally within a `version` target, we need to use `eval`. The way the Makefile was written, `VERSION` was always evaluated. Just because the `ifeq` appeared under `version:` does not mean it is only evaluated when `version` is called.

Reply:
makes sense.